app/jobs/base.rb

```diff
@@ -461,6 +470,50 @@ def self.enqueue_at(datetime, job_name, opts = {})
     enqueue_in(secs, job_name, opts)
   end
 
+  def self.decrease_batch_count(batch_id)
+    batch_key = "batch:#{batch_id}"
+
+    Sidekiq.redis do |redis|
+      if redis.hincrby(batch_key, "ref_count", -1) == 0
+        batch = redis.hgetall(batch_key)
+        redis.del(batch_key)
+
+        # An instance of the same job might run several times (e.g. if it fails
+        # and is retried). When that happens, `batch` might be undefined because
+        # the batch job has already been enqueued and deleted from Redis.
+        Jobs.enqueue(batch["job_name"], JSON.parse(batch["opts"])) if batch["job_name"]
+      end
+    end
+  end
+
+  def self.enqueue_after(job_name, opts = {})
```

Comment:
Let's write some docs for this describing how this should be used and provide examples of its usage.